cdap-common/src/main/java/io/cdap/cdap/common/ArtifactNotFoundException.java

```diff
@@ -35,4 +35,8 @@ public ArtifactNotFoundException(NamespaceId namespace, String name) {
   public ArtifactNotFoundException(ArtifactId artifactId) {
     super("artifact", artifactId.toString());
   }
+
+  public ArtifactNotFoundException(io.cdap.cdap.api.artifact.ArtifactId artifactId) {
+    super("artifact", artifactId.toString());
```

Comment:
Better construct a proto `ArtifactId` and call the other constructor. The main difference is that the api ArtifactId doesn't have namespace in it.